plutus-core/plutus-core/src/PlutusCore/DeBruijn/Internal.hs

```diff
@@ -80,8 +115,14 @@ data NamedDeBruijn = NamedDeBruijn {ndbnString :: !T.Text, ndbnIndex :: !Index}
   deriving stock (Show, Generic, Read)
   deriving anyclass (NFData)
 
--- | A wrapper around nameddebruijn that must hold the invariant of name=`fakeName`.
-newtype FakeNamedDeBruijn = FakeNamedDeBruijn NamedDeBruijn
+{-| A wrapper around `NamedDeBruijn` that *must* hold the invariant of name=`fakeName`.
+
+We do not export the `FakeNamedDeBruijn` constructor: the projection `FND->ND` is safe
+but injection `ND->FND` is unsafe, thus they are not isomorphic.
```

Comment:
I don't know if "unsafe" is exactly the right word here, but I don't have anything better to suggest.